### DISCUSSION TITLE
Expose the PCP from the networking code

### DISCUSSION THREAD

**Discussion by MOZGIII** [n0-computer/iroh#1725] [General]
Hey guys! I was looking around and found your Port Control Protocol implementation. See, I'd really like to use it, and I think I can via the `iroh-net` crate, but I really only want the port mapper functionality.

Would you consider extracting your PCP implementation as a separate crate? I think, for the most flexibility, this part of the dependency tree should look somewhat like this: `iroh-net` -> `portmapper` -> `pcp` - where each of these is a separate crate, and the arrow is a dependency relation, where the crate on the left of the arrow depends on the crate on the right of the arrow.

**Comment by divagant-martian** [comment]
https://crates.io/crates/pcp-client
